src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in driversDB.", e);
+        }
+        deleteAllRelationsToDrivers(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    private void deleteAllRelationsToDrivers(Long carId) {
+        String deleteRelationsQuery = "DELETE FROM cars_drivers "
+                + "WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteRelationsStatement =
+                        connection.prepareStatement(deleteRelationsQuery)) {
+            deleteRelationsStatement.setLong(1, carId);
+            deleteRelationsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete relations between "
+                    + "cars and drivers by car id: " + carId, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete driver with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Optional<Car>> getAllByDriver(Long driverId) {
+        String getAllCarIdsQuery = "SELECT car_id "
+                + "FROM cars_drivers "
+                + "JOIN cars "
+                + "ON id = car_id "
+                + "WHERE driver_id = ?";
+        List<Long> carIds = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarIdsStatement =
+                        connection.prepareStatement(getAllCarIdsQuery)) {
+            getAllCarIdsStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarIdsStatement.executeQuery();
+            while (resultSet.next()) {
+                carIds.add(resultSet.getObject("car_id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all car ids by driver id: " + driverId, e);
+        }
+        List<Optional<Car>> cars = new ArrayList<>();
+        for (Long id: carIds) {
+            cars.add(get(id));
+        }
```

Comment:
I think it is better to get all cars info in getAllCarIdsQuery, and than for every car, get it's drivers